src/mono/sample/wasm/browser2/main.js

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import DotNet from "./dotnet.js"; // User can import DotNet so that there can even potentially be multiple instances in 1 app
+
+// Called when the mono-config.js (soon to be mono-config.json via PR#53606) is loaded
+function onConfigLoaded (config) {
+    // we can do some modification to config at runtime here
+    // then return it
+    // i.e. config.newProp = "test new prop";
+
+    return config;
+}
+
+// Called when Emscripten runtime is loaded
+function onRuntimeInitialized () {
+}
+
+// Called when MONO runtime is loaded and ready or if it errors
+function onMonoRuntimeInitialized (error) {
+    if (error) {
+        // mono didn't load correctly
+        test_exit(1); // test runner related
+        throw(error);
+    } else {
+        // mono loaded and we can start doing stuff with it
+        App.init()
+    }
+}
+
+DotNet.onConfigLoaded = onConfigLoaded; // optional
+DotNet.onRuntimeInitialized = onRuntimeInitialized; // optional
+DotNet.onMonoRuntimeInitialized = onMonoRuntimeInitialized; // required as it acts like the entry point into the program
```

Comment:
Let's make this optional too ? If undefined we could call `App.init()` on behalf of the user.

Reply:
App.init is just what the old browser sample used (see further down the file). `App` is entirely user defined and thus not required by us for then to have. Do you think we should make it required? If so, we can make this callback optional.

I think we should keep it as is because I think it is a bit simpler and easier to understand since they have 3 callbacks instead of 3 callbacks + some mandatory object (that can't be renamed) with a mandatory function in it (that also can't be renamed).